.github/workflows/check-backend-on-pr.yml

```diff
@@ -0,0 +1,114 @@
+name: "Check Backend on Pull Request"
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+    paths:
+      - "backend/**/*.rs"
+      - "backend/**/Cargo.toml"
+      - "backend/**/Cargo.lock"
+      - "backend/rust-toolchain.toml"
+      - "backend/rustfmt.toml"
+      - ".github/workflows/*.yml"
+  push:
+    branches:
+      - develop
+
+# Ensures only one build is run per branch, unless pushing to develop
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/develop' && github.run_number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_ARGS: ${{ github.ref == 'refs/heads/develop' && '--release' || '' }}
```

Comment:
I think it is better to just define per job, which cargo args to use. We don't want to use `--release` for clippy.

```suggestion
```